### PR TITLE
feat(splitio-api-wrapper): implement httpGet function

### PR DESCRIPTION
**Pull request by wendychiang**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -1,20 +1,72 @@
...
+func (binding *SplitioAPIBinding) httpGet(path string, since int) (map[string]interface{}, error) {
+	// initiate a new request
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s%s", binding.splitioAPIUri, path), nil)
+	if err != nil {
+		err := fmt.Errorf("Errored when initiate http new request. %s", err)
+		return map[string]interface{}{}, err
+	}
+
+	// add headers
+	req.Header.Set("Accept", "application/json")
```
Comment (silasbw) [inline api/api.go]: Worth using a higher-level client (*e.g.*, https://github.com/go-resty/resty)?

Comment (wendychiang) [reply api/api.go]: Thanks for the suggestion. yeah it becomes much cleaner.

File: api/api.go [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
+
+	if resp.StatusCode() != http.StatusOK {
+		err = fmt.Errorf("Non-OK HTTP status: %s", resp.Status())
+		return map[string]interface{}{}, err
+	}
+
+	var data map[string]interface{}
+	decoder := json.NewDecoder(strings.NewReader(string(resp.Body())))
+	decoder.UseNumber()
+	decoder.Decode(&data)
```
Comment (rohanmurthy) [inline api/api.go]: Should we add an `if` statement to consider the case where the error returned from this function is not nil? https://golang.org/pkg/encoding/json/#Decoder.Decode

File: api/api.go [hunk 3]
```diff
@@ -1,20 +1,68 @@
...
+		err = fmt.Errorf("Http get request error: %s", err)
+		return map[string]interface{}{}, err
+	}
+
+	if resp.StatusCode() != http.StatusOK {
+		err = fmt.Errorf("Non-OK HTTP status: %s", resp.Status())
+		return map[string]interface{}{}, err
+	}
+
+	var data map[string]interface{}
```
Comment (silasbw) [inline api/api.go]: Can you use https://godoc.org/github.com/go-resty/resty#Response.Result here?

Comment (wendychiang) [reply api/api.go]: Do you mean not using `resp.Body()` in line 57, or making the variable `data` another type?

Comment (wendychiang) [reply api/api.go]: Re: thanks for the suggestion. yeah I can use `resp.Result()` by adding SetResult() first:
```
client.R().SetResult(map[string]interface{}{})
```
but the problem I encounter is that the unmarshal method will return every number as float, but I want an integer.

or I can do `SetResult(map[string]json.RawMessage{})` but that will still require some data manipulation later on.
Do you think make it json.RawMessage{} will be better?

Comment (wendychiang) [reply api/api.go]: Re: merge this first, but can come back to modify it if that's better later on

**Pull request merged**